### PR TITLE
[Feature:Forum] Unread feature to discussion form thread

### PR DESCRIPTION
**Pull request by williamlin6803**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
The current implementation of Submitty, when you click a thread it becomes read to the user but if it a long post you may want to make it unread so that you can read it later. 

### What is the new behavior?
This feature implements what is mentioned above allowing the user to mark a thread as unread if the user wants to go back to the thread later without loosing it. 

IMG_4070

### CODE REVIEW

File: site/app/views/forum/ForumThreadView.php [hunk 1]
```diff
@@ -293,6 +294,7 @@ public function showForumThreads($user, $posts, $unviewed_posts, $threadsHead, $
         }
         else {
             $return = $this->core->getOutput()->renderTwigTemplate("forum/GeneratePostList.twig", [
+                "current_user" = $user,
```
Comment (Kush-Raval) [inline site/app/views/forum/ForumThreadView.php:297]: should be changed to "current_user" => $user,

File: site/app/views/forum/ForumThreadView.php [hunk 2]
```diff
@@ -265,6 +265,7 @@ public function showForumThreads($user, $posts, $unviewed_posts, $threadsHead, $
             }
 
             $return = $this->core->getOutput()->renderTwigTemplate("forum/ShowForumThreads.twig", [
+                "current_user" = $user,
```
Comment (Kush-Raval) [inline site/app/views/forum/ForumThreadView.php:268]: = should become =>

File: site/public/js/forum.js [hunk 3]
```diff
@@ -1217,6 +1217,45 @@ function alterShowMergeThreadStatus(newStatus, course) {
     location.reload();
 }
 
+// eslint-disable-next-line no-unused-vars
+function markUnread(current_thread, current_user) {
+    const url = buildCourseUrl(['forum', 'threads', 'mark_unread']);
```
Comment (Kush-Raval) [inline site/public/js/forum.js:1222]: Shouldn't the url also include thread_id and afterwards mark_unread. 
Example:
image

**Review by Kush-Raval — DISMISSED** [review]
There is an error that is preventing me from testing the code: 
I pointed out changes that I think are creating this error.
Let me know after you fix them so I can re-review.

Edit: Sorry I did not realize this was a draft PR

**Comment by micpap25** [comment]
Thank you for your time and effort! We are closing this PR in favor of #10332 which is more active and current. Please feel free to contribute to that PR if you'd like to continue working on this problem.

**Pull request closed without merging**